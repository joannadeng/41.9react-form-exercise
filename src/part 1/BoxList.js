import React, {useState} from "react";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";
import {v4 as uuid} from 'uuid';

const BoxList = () => {
    const initialState = [
        {id:uuid(), bgcolor:"red", width:50, height:50},
        {id:uuid(), bgcolor:"black", width:50, height:50},
        {id:uuid(), bgcolor:"green", width:50, height:50}
    ]
    
    const [boxList, setBoxList] = useState(initialState);
    const addBox = (newBox) => {
        setBoxList(boxList => [...boxList, {id:uuid(),...newBox}])
    }
    const removeBox = (id) => {
        setBoxList((boxList.filter(box => (box.id !=id))))
    }
    return (
        <div>
            <h3>Box List</h3>
            <NewBoxForm addBox={addBox}/>
            <div>
                {boxList.map(({id,bgcolor,width,height}) => <Box id={id} key={id} removeBox={removeBox} color={bgcolor} width={width} height={height}/>)}
            </div>
        </div>
    )
}

export default BoxList;