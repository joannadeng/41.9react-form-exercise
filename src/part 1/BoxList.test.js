import {render} from '@testing-library/react';
import BoxList from './BoxList';

it ('it should render without crashing', () => {
    render(<BoxList />)
})

it ("should match snapshot", () => {
    const {asFragment} = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot()
})

