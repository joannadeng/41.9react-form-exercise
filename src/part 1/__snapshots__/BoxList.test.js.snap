// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Box List
    </h3>
    <form>
      <label
        for="bgcolor"
      >
        Background Color
      </label>
      <input
        id="bgcolor"
        name="bgcolor"
        placeholder="background color"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        min="30"
        name="width"
        placeholder="width"
        type="number"
        value=""
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        min="30"
        name="height"
        placeholder="height"
        type="number"
        value=""
      />
      <button>
        Submit!
      </button>
    </form>
    <div>
      <div
        class="Box"
        style="background-color: red; width: 50px; height: 50px;"
      >
        <button>
          X
        </button>
      </div>
      <div
        class="Box"
        style="background-color: black; width: 50px; height: 50px;"
      >
        <button>
          X
        </button>
      </div>
      <div
        class="Box"
        style="background-color: green; width: 50px; height: 50px;"
      >
        <button>
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
